src/unistd.rs

```diff
@@ -11,18 +11,126 @@ use std::os::unix::io::RawFd;
 use std::path::{PathBuf};
 use void::Void;
 use sys::stat::Mode;
+use std::fmt;
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
 pub use self::linux::*;
 
+/// User identifier
+///
+/// Newtype pattern around `uid_t` (which is just alias). It prevents bugs caused by accidentally
+/// passing wrong value.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct Uid(uid_t);
+
+impl Uid {
+    pub fn from_raw(uid: uid_t) -> Self {
+        Uid(uid)
+    }
+
+    pub fn current() -> Self {
+        getuid()
+    }
+
+    pub fn effective() -> Self {
+        geteuid()
+    }
+
+    pub fn is_root(self) -> bool {
+        self.0 == 0
+    }
+}
+
+impl From<Uid> for uid_t {
+    fn from(uid: Uid) -> Self {
+        uid.0
+    }
+}
+
+impl fmt::Display for Uid {
```

Comment:
You're right, I was thinking of `Debug`. I don't think we need to have `Display` implemented for these types, but I'm not against it either.